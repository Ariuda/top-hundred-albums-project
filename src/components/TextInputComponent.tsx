import React from 'react';

interface Props {
    value: string;
    placeholder: string;
    onChange: (e: any) => void;
}

export const TextInputComponent: React.FC<Props> = ({ value, placeholder, onChange }) => {

    return (
        <input type="text" className="search-input" value={value} placeholder={placeholder} onChange={onChange} />
    )
};

export default TextInputComponent;