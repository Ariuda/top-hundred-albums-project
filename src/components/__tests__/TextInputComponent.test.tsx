import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { TextInputComponent } from '../TextInputComponent';

const props = {
    value: '',
    placeholder: 'test',
    onChange: () => console.log('test function')
}

it('should render the text input component correctly', () => {
    const wrapper = shallow(<TextInputComponent value={props.value} placeholder={props.placeholder} onChange={props.onChange} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});