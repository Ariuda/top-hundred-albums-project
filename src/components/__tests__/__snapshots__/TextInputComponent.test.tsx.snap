// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the text input component correctly 1`] = `
<input
  className="search-input"
  onChange={[Function]}
  placeholder="test"
  type="text"
  value=""
/>
`;
